src/components/AdminModifyTaskQueue.tsx

```diff
@@ -1,73 +1,82 @@
-import React, { useState, Dispatch, SetStateAction } from 'react';
-import { FaChevronDown } from 'react-icons/fa';
+import React, { useState, Dispatch, SetStateAction, useEffect } from 'react';
+import Select from 'react-select';
+
+interface Options{
+  value: number,
+  label: string
+}
+interface Worker{
+  friendlyName: string,
+  attributes: string,
+  sid: string
+}
+
 
 interface IProps{
   sid: number,
-  taskQueueName: string,
+  taskQueueName: string
   setShowModal: Dispatch<SetStateAction<boolean>>;
   handleDataChange: () => any;
   editType: string;
-
+  workerList: Worker[],
+  assignedWorkers: string[]
 }
 
 const workspaceSid = process.env.NEXT_PUBLIC_WORKSPACE_SID as string
 
-const AdminModifyTaskQueue = ( {sid, taskQueueName, setShowModal, handleDataChange, editType}: IProps)=> {
+const AdminModifyTaskQueue = ( {sid, taskQueueName, setShowModal, handleDataChange, editType, workerList, assignedWorkers}: IProps)=> {
   
-  const [workersToAdd, setWorkersToAdd] = useState('');
-  const [workersToRemove, setWorkersToRemove] = useState('');
-  const [showEditOptions, setShowEditOptions] = useState(false);
-  const [editOptionsValue, setEditOptionsValue] = useState('Add Workers');
+  const [multiSelectOptions, setMultiSelectOptions] = useState<Options[]>([])
+  const [selectedOptions, setSelectedOptions] = useState<Options[]>([])
+
+  const handleChange = (options:any)=>{
+    setSelectedOptions(options)
+  }
   
   async function handleModifyTaskQueue() {
 
     if(editType === "Delete"){
       handleDeleteTaskQueue(sid)
       return
     }
-
-    const getWorkers = await fetch(`/api/workers?workspaceSid=${process.env.NEXT_PUBLIC_WORKSPACE_SID}`,{
-      method:'GET'
-    })
-    const workersResponse = await getWorkers.json()
-    const workersList = workersResponse.workers
-
-    var parsedWorkersToAdd = workersToAdd.split(', ')
-    var parsedWorkersToRemove = workersToRemove.split(', ')
+    if(selectedOptions.length === 0){
+      handleCancel()
+      return
+    }
 
     var addFlag = 0         // Check if add operations were valid
```

Comment:
Nit: can you use `let` instead of `var`. See [this](https://stackoverflow.blog/2019/09/12/practical-ways-to-write-better-javascript/#:~:text=without%20feeling%20guilty.-,let%20and%20const,-For%20most%20of) article